### PR TITLE
bugfix for #1657 "Generate Invoices from Shipments (manual)"

### PR DESCRIPTION
**Pull request by homebeaver**
Hi @yamelsenih - PR as promissed

A note to autoquery, explained in https://github.com/adempiere/adempiere/issues/1657#issuecomment-383010457 : 

- this is like in smartbrowser - default method in interface `FormPanel `needs java 8 
- I'll implement autoquery also in similar classes 
 `VInOutGen` / "Generate Shipment (manual)",
 `VInvoiceGen` /"Generate Invoice (manual)"  and 
`VInOutInvoiceGen `/ "Genarate Shipments & Invoices (manual)" 
, see https://github.com/adempiere/adempiere/issues/1652#issue-313648036

enjoy

### CODE REVIEW

File: client/src/org/adempiere/apps/form/InvoiceGenFromShipment.java [hunk 1]
```diff
@@ -276,6 +287,32 @@ public String generate(IStatusBar statusBar, KeyNamePair docTypeKNPair, String d
 			return info;
 		}
 		
+		if(m_AD_Org_ID==null) {
+			// ??? TODO 
+		} else { // damit werden aber Re für alle DB erstellt! wenn kein m_C_BPartner_ID
+			para = new MPInstancePara(instance, 30);
+			para.setParameter("AD_Org_ID", (Integer)m_AD_Org_ID); // eigentlich int 
+			if (!para.save()) {
+				String msg = "No AD_Org_ID Parameter added";  //  not translated
```
Comment (yamelsenih) [inline client/src/org/adempiere/apps/form/InvoiceGenFromShipment.java]: Hi @homebeaver, have it a Translation? is better if exist a Msg.translate methiod here

File: base/src/org/adempiere/process/InvoiceGenerateFromShipment.java [hunk 2]
```diff
@@ -48,9 +49,33 @@
...
  *		<a href="https://github.com/adempiere/adempiere/issues/514">
  * 		@see FR [ 514 ] Error in process to Generate invoices from shipments</a>
+ *  @author https://github.com/homebeaver
+ *	@see <a href="https://github.com/adempiere/adempiere/issues/1657"> 
+ * 		  bug in swing client "Generate Invoices from Shipments (manual)"</a>
  */
-public class InvoiceGenerateFromShipment extends InvoiceGenerateFromShipmentAbstract {
+public class InvoiceGenerateFromShipment extends SvrProcess {
 	
+	/**	Manual Selection		*/
```
Comment (yamelsenih) [inline base/src/org/adempiere/process/InvoiceGenerateFromShipment.java]: Hi @homebeaver it is unnecessary with new approach of Abstract class for process

File: base/src/org/adempiere/process/InvoiceGenerateFromShipment.java [hunk 3]
```diff
@@ -48,9 +49,33 @@
  *  @author Yamel Senih, ysenih@erpcya.com, ERPCyA http://www.erpcya.com
  *		<a href="https://github.com/adempiere/adempiere/issues/514">
  * 		@see FR [ 514 ] Error in process to Generate invoices from shipments</a>
+ *  @author https://github.com/homebeaver
+ *	@see <a href="https://github.com/adempiere/adempiere/issues/1657"> 
+ * 		  bug in swing client "Generate Invoices from Shipments (manual)"</a>
  */
-public class InvoiceGenerateFromShipment extends InvoiceGenerateFromShipmentAbstract {
+public class InvoiceGenerateFromShipment extends SvrProcess {
```
Comment (yamelsenih) [inline base/src/org/adempiere/process/InvoiceGenerateFromShipment.java]: It is old approach

File: base/src/org/adempiere/process/InvoiceGenerateFromShipmentAbstract.java [hunk 4]
```diff
@@ -1,173 +0,0 @@
-/******************************************************************************
```
Comment (yamelsenih) [inline base/src/org/adempiere/process/InvoiceGenerateFromShipmentAbstract.java]: This class is automatically generated from ADempiere is better use InvoiceGenerateFromShipment.java class

File: base/src/org/adempiere/process/InvoiceGenerateFromShipment.java [hunk 5]
```diff
@@ -65,23 +90,26 @@
 	 */
 	protected void prepare() {
 		log.config("");
-		super.prepare();
+		p_Selection = getParameterAsBoolean("Selection");
```
Comment (yamelsenih) [inline base/src/org/adempiere/process/InvoiceGenerateFromShipment.java]: Is unnecessary with new approach

File: client/src/org/adempiere/apps/form/InvoiceGenFromShipment.java (hunk 1 above)
Comment (homebeaver) [reply client/src/org/adempiere/apps/form/InvoiceGenFromShipment.java]: yes, you are right. This is originally old code, not translated. I did not change it at this point.

**Comment by homebeaver** [comment]
@yamelsenih , tkx for comment.
I overlooked #326 because we have 3.9.0 in production - and this is my baseline. So I do not work with the actual develop branch. When a fix is needed I look for it and get it with cherry-picking. 
So this is the reason for the fix  of `InvoiceGenerateFromShipment `with "old" approach. But it works.
Maybe I chench the pull to the new approach in the next days - but low priority.
regards

**Comment by yamelsenih** [comment]
Hi @homebeaver i close it pull request, is nice your contribution but is better if you use current tools for your changes, we hope news about it changes over 3.9 develo.

Best regard

**Comment by yamelsenih** [comment]
Maybe you can see the new Smart Browser for generate Shipments and Invoice from Order, the current form will should be deprecated for next versions

**Comment by homebeaver** [comment]
Hi @yamelsenih ,
by declining this pull you "throw the baby out with the bath water". But nevertheless I will send a new pull request which uses `InvoiceGenerateFromShipmentAbstract` - the new functionality.

regards and hold the line

**Pull request closed without merging**